src/main/java/com/ericsson/ei/controller/SubscriptionControllerImpl.java

```diff
@@ -55,133 +54,132 @@
-                subscriptionService.addSubscription(subscription);
-                LOG.info("Subscription :" + subscription.getSubscriptionName() + " Inserted Successfully");
-                subscriptionResponse.setMsg("Inserted Successfully");
-                subscriptionResponse.setStatusCode(HttpStatus.OK.value());
-                subResponse = new ResponseEntity<>(subscriptionResponse, HttpStatus.OK);
 
-            } else {
-                LOG.error("Subscription :" + subscription.getSubscriptionName() + " already exists");
-                subscriptionResponse.setMsg("Subscription already exists");
-                subscriptionResponse.setStatusCode(HttpStatus.BAD_REQUEST.value());
-                subResponse = new ResponseEntity<>(subscriptionResponse, HttpStatus.BAD_REQUEST);
+                if (!subscriptionService.doSubscriptionExist(subscriptionName)) {
+                    subscription.setUserName(user);
+                    subscription.setCreated(Instant.now().toEpochMilli());
+                    subscriptionService.addSubscription(subscription);
+                    LOG.debug("Subscription inserted successfully :: " + subscriptionName);
+                } else {
+                    LOG.error("Subscription already exists :: " + subscriptionName);
+                    errorMap.put(subscriptionName, "Subscription already exists");
+                }
+            } catch (Exception e) {
+                LOG.error("Error on subscription " + subscriptionName + ", " + e.getMessage());
+                errorMap.put(subscriptionName, e.getMessage());
             }
-        }
-        return subResponse;
+        });
+        return (errorMap.isEmpty()) ? new ResponseEntity<>(HttpStatus.OK) : new ResponseEntity<>(getSubscriptionResponseList(errorMap), HttpStatus.BAD_REQUEST);
     }
 
     @Override
     @CrossOrigin
-    @ApiOperation(value = "Returns the subscription rules for given subscription name")
-    public ResponseEntity<List<Subscription>> getSubscriptionById(@PathVariable String subscriptionName) {
-        List<Subscription> subscriptionList = new ArrayList<>();
-        try {
-            LOG.info("Subscription :" + subscriptionName + " fetch started");
-            subscriptionList.add(subscriptionService.getSubscription(subscriptionName));
-            LOG.info("Subscription :" + subscriptionName + " fetched");
-            return new ResponseEntity<>(subscriptionList, HttpStatus.OK);
-        } catch (SubscriptionNotFoundException e) {
-            LOG.error("Subscription :" + subscriptionName + " not found in records");
-            return new ResponseEntity<>(subscriptionList, HttpStatus.BAD_REQUEST);
-        }
+    @ApiOperation(value = "Returns the subscriptions for given subscription names separated by comma")
+    public ResponseEntity<GetSubscriptionResponse> getSubscriptionById(@PathVariable String subscriptionName) {
+        List<String> subscriptionNames = Arrays.asList(subscriptionName.split(","));
```

Comment:
Change "subscriptionName" to a more suitable since it is a list in this case -> e.g. "subscriptionNames" or "subscriptionaNameList".
"getSubscriptionById" function name should be renamed to e.g. "getSubscriptionByIds" or "getSubscriptionByNames".

Reply:
yeah, I will check one more time what I missed and fix it.